hash_reader.go

```diff
@@ -133,43 +194,92 @@ func (hashReader *hashReader) TimeValue() (time.Time, error) {
 }
 
 func (hashReader *hashReader) StringValue() (string, error) {
-	panic("implement me")
 	return hashReader.ionReader.StringValue()
 }
 
 func (hashReader *hashReader) ByteValue() ([]byte, error) {
-	panic("implement me")
 	return hashReader.ionReader.ByteValue()
 }
 
 func (hashReader *hashReader) Sum(b []byte) []byte {
-	panic("implement me")
+	return hashReader.hasher.sum()
+}
+
+func (hashReader *hashReader) traverse() error {
+	for hashReader.Next() {
```

Comment:
There is a pending discussion on what `Next()` should return in ion-go that may affect this code here.